lisa/wlgen/workload.py

```diff
@@ -80,15 +81,29 @@ def __init__(self, target, name, res_dir=None):
         self.name = name
         self.command = None
         self.output = ""
-        self.run_dir = self.target.working_directory
 
+        self.target.execute(
+            "mkdir -p {}/lisa/wlgen".format(target.working_directory))
```

Comment:
Would need some quoting and `target.join`:
```
import shlex
"mkdir -p {}".format(shlex.quote(target.path.join(target.working_directory, "lisa", "wlgen")))
```

Maybe worth adding a helper to devlib's target to create folders on the target as it seems a quite basic/common case.

Reply:
Ok, let's keep the devlib integration for another time ;)
Also because I notice devlib is already using the pattern above in many different places and that patch will mean a big refactoring of devlib to just add a small utility method.

I'll use the same devlib pattern and we should be good on lisa too ;)